tests/testthat/test-export.R

```diff
@@ -7,16 +7,47 @@ testthat::test_that("Listing print correctly", {
   })
 })
 
+testthat::test_that("key columns repeat with export_as_txt", {
+  # pre-processing and ordering
+  tmp_data <- ex_adae %>%
+    dplyr::slice(1:30) %>%
+    dplyr::distinct(USUBJID, AGE, BMRKR1, .keep_all = TRUE)
+
+  lsting <- as_listing(tmp_data,
+                       key_cols = c("USUBJID", "AGE"),
+                       disp_cols = character()
+  ) %>%
+    add_listing_col("BMRKR1", format = "xx.x")
+
+  listing_exp <- suppressMessages(export_as_txt(lsting, lpp = 4, verbose = TRUE, page_break = "\n",
+                                                rep_cols = length(get_keycols(lsting))))
```

Comment:
@shajoezhu I think the second direct regression test is not needed. This is covering the same bug, I would add as comment the number of the issue